src/library/Avatar/Avatar.js

```diff
@@ -28,17 +27,23 @@ export default class Avatar extends Component<AvatarProps> {
           ) : (
             <span>{child}</span>
           );
-      } else if (
-        child.type &&
-        child.type.displayName &&
-        child.type.displayName.indexOf('Icon') !== -1
-      ) {
-        icon = cloneElement(child, {
-          size: ICON_SIZE[size || Avatar.defaultProps.size]
-        });
-      } else {
-        noBackground = true;
+
+        return;
```

Comment:
The early `return` exits make the diff look like you're changing more than you are.

Even though I promote the use of early exit in our own code since this is Mineral UI and we're trying to make the least changes possible... Can you just revert and keep the `if/else if/else` structure to simplify the diff and avoid confusion later.

To be clear, the logic itself seems fine. It's just the move from `if/else if` to early exit I'm suggesting we change back.